src/relion/_parser/class2D.py

```diff
@@ -52,7 +52,7 @@ def __iter__(self):
         return iter(self.jobs)
 
     def __len__(self):
-        return sum(1 for d in self._basepath.iterdir() if d.is_dir())
+        return sum(1 for d in self.jobs)
```

Comment:
this can then just be `len(self.jobs)` as that is guaranteed to be a list.

(same for the other classes)

Reply:
(Which does raise the question whether it's worth to refactor out a superclass)